### ISSUE TITLE
"Tab" key Not Working when cursor in Bullet Point

### ISSUE THREAD

**Issue by OdyAsh** [Pierrad/obsidian-github-copilot#38]
**Describe the bug**
If I assign any "Tab" related key to "Accept Suggestion" HotKey, it doesn't get registered when the cursor is on a bullet point or a numbered list.

**To Reproduce**
1. Setup the "Accept Suggestion" HotKey to any "Tab" related key
2. Make the mouse's cursor stand on any bullet point or numbered list
3. Wait for a suggestion to pop-up
4. Click Tab
5. See that the text is not accepted.

**Expected behavior**
The text to be accepted.

**Obsidian and environment (please complete the following information):**
 - OS: Windows 11 
![image](https://github.com/user-attachments/assets/c7c837e0-e3af-483d-b118-a82dd378fc83)

 - Obsidian version: 1.6.7
 - Plugin version [e.g. 1.0.0] : 1.0.11

**Comment by weber8thomas** [comment]
Same issue on MacOS Sonoma

**Comment by Pierrad** [comment]
Hi @OdyAsh, 

Do you have any special plugins?
Some users has the same problem and it was always related to other plugin interfering with the key mapping (since `1.0.6`, before that it was buggy but you're on `1.0.11` so you should be good)

**Comment by OdyAsh** [comment]
Hi there!

I have a lot of plugins actually 😅, but I checked the hotkeys menu and didn't see any overlapping hotkeys with the "Tab" key. Moreover, I disabled the plugins which would logically cause an issue (Text editor and Code Editor Shortcuts), yet the issue persists :[

My plugins for reference:

![image](https://github.com/user-attachments/assets/071a58cd-197c-4db7-9b09-e98ab2fabdbb)

![image](https://github.com/user-attachments/assets/cf11197c-c00c-43bc-a068-c0972d698c10)

![image](https://github.com/user-attachments/assets/5f412ba2-3574-480f-bd22-1b706b9cf326)

![image](https://github.com/user-attachments/assets/6d5a12db-2435-4970-83b8-57d80cddf4a7)

**Comment by erik-buchholz** [comment]
@OdyAsh 
I had exactly the same issue. 
Then, I deactivated all other plug-ins (except this Github Copilot) and activated them one by one to find the culprit. 
Now, all my plug-ins are activated again and that "Tab" completion works even in lists.
I restarted Obsidian multiple times, and it still works.
Maybe this will do the trick for you, too. Or maybe "Reload Plugins" in the settings is already sufficient to fix the bug.

**TL;DR: 
Deactivating and re-activating all other plug-ins fixed the problem.**

**Comment by jaquesgrobler** [comment]
More specifically, Outliner was the culprit for me -
I deactivated and re-activated 'Outliner' and now tab works correctly for accepting completions as with @erik-buchholz solution. HTH

**Comment by erik-buchholz** [comment]
@jaquesgrobler 
I can confirm that Outliner is causing the problem. Unfortunately, the bug reappears every other day, but reactivating Outliner resolves it. 
Looking @OdyAsh screenshots, he also uses Outliner, so this is most likely the cause. 

Right now, the bug isn't occurring on my system, so I cannot confirm it, but the Outliner has an option to "Enhance the Tab key" in the plug-in settings. Potentially deactivating this feature might permanently resolve the problem.

**Comment by Pierrad** [comment]
Hi, thanks for your investigation, I will keep this issue open for potential new people.

**Comment by GauSyu** [comment]
Similar issue for me. But I only have the following plugins: Github Copilot, Latex Suite, PDF++, TikZJaX, and ZotLit. The issue happens in math mode, but I have no idea why.

**Comment by Pierrad** [comment]
@GauSyu Indeed, if I try to add a bullet point in LateX Math mode, the "Tab" key is not working. But if I try to change the key assigned to the "accept" command, it works.

I tried to debug and it seems that the LateX Math mode prevents my KeyWatcher to detect the specific "Tab" keypress in a bullet point. Maybe the LateX plugin has the priority over my plugin, I don't know :/

**Comment by astraetech** [comment]
Hi all,

Same problem. I disabled all community plugins (incl Outliner) and the problem still persists. Instead of accepting the Copilot suggestions, Tab just indents the numbered/bullet point. I tried changing the "Accept suggestion" shortcut but that had no effect at all. Latest Obsidian and the Copilot for Obsidian version.
